ksqldb-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -295,25 +277,45 @@ public boolean deleteType(final String name) {
   private static final class SourceInfo {
 
     private final DataSource source;
-    private final ReferentialIntegrityTableEntry referentialIntegrity;
+    private final Set<SourceName> linkedSources = new ConcurrentHashSet<>();
+    private final Set<SourceName> referentialSources = new ConcurrentHashSet<>();
```

Comment:
can we document what the difference between these two are?

Reply:
Done